### DISCUSSION TITLE
Timed out in bb8 when connecting using IP

### DISCUSSION THREAD

**Discussion by 0100-0100** [sfu-db/connector-x#403] [Q&A] [answered]
Hi,

While trying to connect to a MS SQL Server Database on my local network I get an Error message indicating Timed out in bb8,

I have not been able to identify what exactly is causing the connection attempt to time out, and I have tried to use the urllib.parse.urlparse() method suggested on the internet but I still get the same result when attempting to connect to my local database server,

---

pyodbc and Alchemy have no issue with the connection using the IP address on my LAN so I'm stuck using connectorx for the cloud connections I need only and other connector when reading my local DB.

Here's how the connection string looks like, I changed my actual user for placeholder USER.

`mssql://USER:FullStack01.!@192.168.40.6/biometrics?encrypt=true&trusted_connection=true&TrustServerCertificate=Yes`

---

Any help on Identifying what the issue might be will be greatly appreciated,

I have looked a bit on the source code of connectorx but the only thing I managed to identify is that the error comes from the _read_sql() method in connectorx/__init__.py on line 224 but provides no details as to what is causing the timeout.

Here's a sample code I wrote to give a better idea of what I'm doing:

> import connectorx as cx
> url = 'mssql://USER:FullStack01.!@192.168.40.6/biometrics?encrypt=true&trusted_connection=true&TrustServerCertificate=Yes'
> query = 'SELECT * FROM dbo.biometrics'
> df = cx.read_sql(url, query)

Below the full traceback:
>Traceback (most recent call last):
>  File "<stdin>", line 1, in <module>
>  File "/home/ubuntu/.local/lib/python3.8/site-packages/connectorx/__init__.py", line 224, in read_sql
>    result = _read_sql(
> RuntimeError: Timed out in bb8

**Comment by wangxiaoying** [comment, marked as answer]
Hi @DiegoCol93 , have you tried to remove the url parameters like `encrypt`, `trusted_connection`? Also we don't support parameter like `TrustServerCertificate`.

**Reply by 0100-0100** [reply]
Thanks a lot, definitely helped and indeed worked, Now I only need to wait for UID strings partition support or add an id on the table I'm using but I did see the improvement using this connector, kudos @wangxiaoying
